source/common/rate_limiter_impl.cc

```diff
@@ -81,38 +78,45 @@ void LinearRateLimiter::releaseOne() {
   acquired_count_--;
 }
 
-DelegatingRateLimiter::DelegatingRateLimiter(Envoy::TimeSource& time_source,
-                                             RateLimiterPtr&& rate_limiter,
-                                             RateLimiterDelegate random_distribution_generator)
-    : random_distribution_generator_(std::move(random_distribution_generator)),
-      time_source_(time_source), rate_limiter_(std::move(rate_limiter)) {}
+DelegatingRateLimiterImpl::DelegatingRateLimiterImpl(
+    RateLimiterPtr&& rate_limiter, RateLimiterDelegate random_distribution_generator)
+    : ForwardingRateLimiterImpl(std::move(rate_limiter)),
+      random_distribution_generator_(std::move(random_distribution_generator)) {}
 
-bool DelegatingRateLimiter::tryAcquireOne() {
+bool DelegatingRateLimiterImpl::tryAcquireOne() {
+  const auto now = timeSource().monotonicTime();
```

Comment:
Is there a slight functional change here?  I'm not 100% sure I understand how monotonicTime() works here, but because you're pulling this value before tryAcquireOne here, if tryAcquireOne were to ever have a delay for any reason (even if only in certain impls), what `now` is capturing here is not equivalent to what `now` was capturing before.

If you don't think this is a legitimate concern, I will completely defer to you on this.

Reply:
Nice, catch; I backed this out. 
The behavioral change is quite small but as the PR description promised to not introduce these type of changes, this shouldn't go in here - and probably there is no value in making the change at all.

Reply:
Thank you!